### ISSUE TITLE
[JOSS] @rcannood's feedback for the JOSS submission

### ISSUE THREAD

**Issue by rcannood** [jhayer/baargin#13]
Hi all!

I'm currently reviewing this repository's JOSS submission (https://github.com/openjournals/joss-reviews/issues/5397). I'll update this issue throughout the review process and will notify you when I'm finished with the review of the first version of the manuscript :)

Robrecht

-----

# Functionality

## Installation

- [x] #34

- [x] #44
  ```bash
  nextflow run jhayer/baargin -r 0.1.0 --help
  ```

# Documentation

## Statement of need

- [x] #19

## Installation instructions

- [x] #35

## Functionality documentation

- [x] #21

## Automated tests

- [x] #47

## Community guidelines

- [x] #24

# Software paper

## Summary

- [x] #14

## State of the field

- [x] #29

## Quality of writing

- [x] #18
- [x] #16

**Comment by rcannood** [comment]
Alright, I went through the checklist. @jhayer if you disagree with any of the comments, feel free to discuss.

It would be fantastic if you could tackle each comment in this issue through a separate PR (whenever possible), because that makes my life so much easier in checking whether the changes has resolved my concerns ;)

**Comment by jhayer** [comment]
Thank you @rcannood, that's great. I will go through it ASAP. I am also waiting for comments from the second reviewer.

**Comment by rcannood** [comment]
Hi @jhayer !

Thanks for the updates, things are looking great!

Below are some minor issues which still need to be resolved.


## Functionality

### Installation - Database prerequisites - Manual 

The documentation states:

> Prerequisites The use of the script download_db.py requires python and python modules.
> Prerequisite: Python>=3.8.0  
> ```
> pip install pyyaml gitpython requests biopython>=1.78 numpy>=1.22
> ```

- [x] Please quote `biopython>=1.78` and `numpy>=1.22` to achieve the desired effects

### Warnings while running

Running the pipeline produces a _lot_ of warnings.

```
WARN: A process with name 'quast_contigs_only2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_quast2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'busco2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_busco2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'amrfinderplus2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'amrfinderplus_no_db2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'amrfinderplus_no_species2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'amrfinderplus_no_species_no_db2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'card_rgi2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'plasmidfinder2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'platon2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'platon_json2tsv2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_platon2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'mefinder2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'mlst2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_mlst2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_amrfinder2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_amrfinder_no_species2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_plasmidfinder2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: A process with name 'compile_card2' is defined more than once in module script: baargin/main.nf -- Make sure to not define the same function as process
WARN: Access to undefined parameter `reads_folder` -- Initialise it to a default value eg. `params.reads_folder = some_value`
```

* [x] The one about the parameters you can resolve by replacing `if(params.reads_folder){` with `params.contains("reads_folder") && params.reads_folder`.

* [x] The one about defining a process with name more than once you can resolve by swapping the order of the aliases.
  - From: `include {quast_contigs_only; quast_contigs_only as quast_contigs_only2} from './modules/quast.nf'`
  - To: `include {quast_contigs_only as quast_contigs_only2; quast_contigs_only} from './modules/quast.nf'`

### Minor improvements


* [x] Please add the following to the git ignore:

    ```
    # nextflow-related ignores
    /work
    /.nextflow*
    /db
    /results_test
    ```

## Software paper

- [x] Don't forget to rerender the manuscript PDF in the JOSS issue :)

## Open issues

Have you thought about how to tackle these two remaining items?

* [x] #44
* [x] #47

**Comment by jhayer** [comment]
Hi @rcannood,
I think we have now addressed all your comments. Thanks again for your useful input.

**Comment by rcannood** [comment]
Thanks @jhayer ! I feel like all of my concerns have been addressed :+1:
